internal/controller/managedzones.go

```diff
@@ -0,0 +1,50 @@
+package controller
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/kuadrant/dns-operator/api/v1alpha1"
+	"github.com/kuadrant/dns-operator/internal/provider"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func (r *ManagedZoneReconciler) publishManagedZone(ctx context.Context, managedZone *v1alpha1.ManagedZone) error {
```

Comment:
I kinda want to put these on their own type so they have no access to the client